tests/sklearn/test_sklearn_autolog.py

```diff
@@ -672,3 +682,55 @@ def test_autolog_does_not_throw_when_mlflow_logging_fails(func_to_fail):
 
         model.fit(X, y)
         mock_func.assert_called_once()
+
+
+@pytest.mark.parametrize("data_type", [pd.DataFrame, np.array])
+def test_autolog_logs_signature_and_input_example(data_type):
+    mlflow.sklearn.autolog()
+
+    X, y = get_iris()
+    X = data_type(X)
+    y = data_type(y)
+    model = sklearn.linear_model.LinearRegression()
+
+    with mlflow.start_run() as run:
+        model.fit(X, y)
+        model_path = os.path.join(run.info.artifact_uri, MODEL_DIR)
+
+    model_conf = get_model_conf(run.info.artifact_uri)
+    input_example = _read_example(model_conf, model_path)
+    pyfunc_model = mlflow.pyfunc.load_model(model_path)
+
+    assert model_conf.signature == infer_signature(X, model.predict(X[:5]))
+    np.testing.assert_array_equal(pyfunc_model.predict(input_example), model.predict(X[:5]))
```

Comment:
@harupy I tweaked this test case slightly to verify that the input example can be used to evaluate the pyfunc representation of the model. The example can be loaded as a dataframe using the `mlflow.models.utils._read_example` utility function.

Reply:
makes sense! Thanks!